vowpalwabbit/cbify.cc

```diff
@@ -474,18 +474,16 @@ base_learner* cbifyldf_setup(options_i& options, vw& all)
   data->all = &all;
   data->use_adf = true;
 
-  if (!options.was_supplied("cb_explore_adf"))
-  {
-    options.insert("cb_explore_adf", "");
-  }
-  options.insert("cb_min_cost", std::to_string(data->loss0));
-  options.insert("cb_max_cost", std::to_string(data->loss1));
+  options.ensure_default_dependency("cb_explore_adf");
+
+  options.require("cb_min_cost", std::to_string(data->loss0));
+  options.require("cb_max_cost", std::to_string(data->loss1));
```

Comment:
Whats the difference between `ensure_default_dependency` and `require` here?

Reply:
after doing the refactor, its becoming obvious that it was the same thing after all. I didn't do the full merging of both because I wanted to think about it for sure.

Reply:
I guess it depends if we need to build custom stacks that require this functionality if not they can be merged in a single one.